### PR TITLE
Move schema to a JSON file

### LINKED ISSUE #20: Define schemas in JSON files instead of Python code
https://github.com/aodn/aodn-netcdf-tools/pull/3#discussion_r219022116

e.g. to allow re-use in non-Python applications.

### PR DESCRIPTION
**Pull request by mhidas**
Fixes #20

### CODE REVIEW

**Comment by mhidas** [comment]
@lwgordonimos Following your suggestion (https://github.com/aodn/aodn-netcdf-tools/pull/3#discussion_r219022116)

**Comment by mhidas** [comment]
One thing I'm not quite sure about is if we eventually want to use the schema from other code, it should probably be in a separate **schema** directory outside **ncwriter**, but it would still need to be included in the package. Can we still do that in setup.py with the  `package_data` parameter? (e.g. like [here](https://github.com/aodn/python-aodncore/blob/master/setup.py#L54))

File: ncwriter/schema.py [hunk 1]
```diff
@@ -26,77 +28,27 @@
                                       default_types=types
                                       )
 
+TEMPLATE_SCHEMA_JSON = os.path.join(os.path.dirname(__file__), 'template_schema.json')
```
Comment (lwgordonimos) [inline ncwriter/schema.py]: Maybe look at the package_data mechanism for including non-Python files?

Comment (mhidas) [reply ncwriter/schema.py]: Yep, I'm looking into it.

In terms of accessing the including file using setuptoos functions, [this](https://setuptools.readthedocs.io/en/latest/pkg_resources.html?highlight=data#overview) seems to suggest they rely on having "egg-info" metadata directories. Will that be a problem, given that we're using a wheel to install the package?

**Comment by mhidas** [comment]
See last commit for adding the template JSON file as package data.

Looks like we have the same Travis issue as in aodncore...

**Pull request merged**